src/contrib/mlir/core/ngraph_dialect/dialect.cpp

```diff
@@ -98,16 +97,15 @@ mlir::Type NGraphOpsDialect::parseEltType(mlir::DialectAsmParser& parser) const
             parser.emitError(loc, "Unexpected nGraph integer type: " + origTypeStr);
         }
 
+        auto signedness = isSigned ? NGIntegerType::SignednessSemantics::Signed
+                                   : NGIntegerType::SignednessSemantics::Unsigned;
+
         switch (width)
         {
-        case 8:
-            return isSigned ? NGIntegerType::getInt8(context) : NGIntegerType::getUInt8(context);
-        case 16:
-            return isSigned ? NGIntegerType::getInt16(context) : NGIntegerType::getUInt16(context);
-        case 32:
-            return isSigned ? NGIntegerType::getInt32(context) : NGIntegerType::getUInt32(context);
-        case 64:
-            return isSigned ? NGIntegerType::getInt64(context) : NGIntegerType::getUInt64(context);
+        case 8: return NGIntegerType::get(8, signedness, context);
```

Comment:
Remove the switch. Do `NGIntegerType::get(width, signedness, context)`

Reply:
Good call.  Will do.